fastchat/serve/monitor/topic_clustering.py

```diff
@@ -218,18 +233,27 @@ def get_cluster_info(texts, labels, topk_indices):
     )
     parser.add_argument("--show-top-k", type=int, default=200)
     parser.add_argument("--show-cut-off", type=int, default=512)
+    parser.add_argument("--save-embeddings", action="store_true")
+    parser.add_argument("--embeddings-file", type=str, default=None)
     args = parser.parse_args()
 
     num_clusters = args.num_clusters
     show_top_k = args.show_top_k
     show_cut_off = args.show_cut_off
 
-    texts = read_texts(
-        args.input_file, args.min_length, args.max_length, args.english_only
-    )
-    print(f"#text: {len(texts)}")
+    if args.embeddings_file is None:
+        texts = read_texts(
+            args.input_file, args.min_length, args.max_length, args.english_only
+        )
```

Comment:
This needs to be move out from `if`. otherwise if you load embedding from file, `texts` won't be correctly initialized